x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -635,6 +655,42 @@ protected void createCheckpoint(ActionListener<Void> listener) {
             ));
         }
 
+        @Override
+        public boolean sourceHasChanged() {
```

Comment:
It would be great if there was a way to move this to `onStart` or something so it is executed off of the scheduler thread.

Reply:
the problem is that I do not want to start the indexer if there are no changes, that's why I have this "quick check" in place. We need to measure the costs of this, I agree that blocking the scheduler thread is a concern.